### PR TITLE
[Server] Documented manifest endpoints with apidoc

### PR DESCRIPTION
**Pull request by BourgoisMickael**

### CODE REVIEW

File: server/src/action/manifest/create.js [hunk 1]
```diff
@@ -2,6 +2,67 @@ const AbstractAction = require('../abstract')
 const { body } = require('express-validator/check')
 const { sanitizeBody } = require('express-validator/filter')
 
+/**
+ * @api {post} /api/manifest Create
+ * @apiVersion 1.0.0
+ * @apiGroup Manifest
+ * @apiName ManifestCreate
+ *
+ * @apiDescription Create a manifest to compile
```
Comment (Arignir) [inline server/src/action/manifest/create.js]:
```suggestion
 * @apiDescription Create a new manifest that can be used to compile a package
```

File: server/src/action/manifest/create.js [hunk 2]
```diff
@@ -2,6 +2,67 @@ const AbstractAction = require('../abstract')
...
+ *
+ * @apiParam  (Body) {String{1..255}}  name    Name of the manifest
+ * @apiParam  (Body) {String{1..}}  content Content of the manifest
+ *
+ * @apiParamExample {json} Request Body
+ * {
+ *    "name": "bonjour",
```
Comment (Arignir) [inline server/src/action/manifest/create.js]: Please avoid french word as example, and use realistic content (like `gcc`, `glibc` whatever)

File: server/src/action/manifest/create.js [hunk 3]
```diff
@@ -2,6 +2,67 @@ const AbstractAction = require('../abstract')
...
+ * @apiSuccess  {Object[]}  history               List of content (last updated first)
+ * @apiSuccess  {Integer}   history.id            Id of the content
+ * @apiSuccess  {Integer}   history.manifest_id   Id of the manifest
+ * @apiSuccess  {String}    history.content       The content of the manifest
+ * @apiSuccess  {String}    history.edition_date  Date of content edition
+ *
+ * @apiSuccessExample {json} Response
+ * {
+ *    "id": 56,
+ *    "name": "to_compile",
```
Comment (Arignir) [inline server/src/action/manifest/create.js]: Would be great to be consistent with the example above, so it feels like it is its response.

File: server/src/action/manifest/create.js [hunk 4]
```diff
@@ -2,6 +2,67 @@ const AbstractAction = require('../abstract')
...
+ *          "manifest_id": 56,
+ *          "content": "{{content of a python file manifest}}"
+ *          "edition_date": "2019-07-19T22:25:10.370Z"
+ *      }
+ *    ]
+ * }
+ *
+ * @apiError  {String}    message        Error message
+ * @apiError  {Object[]}  errors         List of error details
+ * @apiError  {String}    errors.param   Request parameter in error
```
Comment (Arignir) [inline server/src/action/manifest/create.js]:
```suggestion
 * @apiError  {String}    errors.param   The request parameter that caused the error
```

File: server/src/action/manifest/create.js [hunk 5]
```diff
@@ -2,6 +2,67 @@ const AbstractAction = require('../abstract')
...
+ * @apiError  {string}    errors.detail  Detail about the parameter in error
```
Comment (Arignir) [inline server/src/action/manifest/create.js]:
```suggestion
 * @apiError  {string}    errors.detail  Details about the parameter that caused the error
```

File: server/src/action/manifest/create.js [hunk 6]
```diff
@@ -2,6 +2,67 @@ const AbstractAction = require('../abstract')
...
+ * @apiError  {string}    errors.value   Value of the parameter provided
```
Comment (Arignir) [inline server/src/action/manifest/create.js]:
```suggestion
 * @apiError  {string}    errors.value   Value provided for the invalid parameter
```

File: server/src/action/manifest/get.js [hunk 7]
```diff
@@ -1,6 +1,66 @@
 const AbstractAction = require('../abstract')
 const { param } = require('express-validator/check')
 
+/**
+ * @api {get} /api/manifest/:id Get
+ * @apiVersion 1.0.0
+ * @apiGroup Manifest
+ * @apiName ManifestGet
+ *
+ * @apiDescription Retrieve a manifest
```
Comment (Arignir) [inline server/src/action/manifest/get.js]:
```suggestion
 * @apiDescription Fetch a manifest's content and metadata
```

File: server/src/action/manifest/create.js [hunk 8]
```diff
@@ -2,6 +2,67 @@ const AbstractAction = require('../abstract')
...
+ * @apiParam  (Body) {String{1..255}}  name    Name of the manifest
+ * @apiParam  (Body) {String{1..}}  content Content of the manifest
+ *
+ * @apiParamExample {json} Request Body
+ * {
+ *    "name": "bonjour",
+ *    "content": "{{content of a python file manifest}}"
+ * }
+ *
+ * @apiSuccess  {Integer}   id                    Id of the manifest
```
Comment (Arignir) [inline server/src/action/manifest/create.js]:
```suggestion
 * @apiSuccess  {Integer}   id                    ID of the manifest
```

(This mistake is present a couple more time in the PR)

File: server/src/action/manifest/get.js [hunk 9]
```diff
@@ -1,6 +1,66 @@
...
+ *
+ * @apiParam  (Params) {Integer}  id  Id of the manifest
+ *
+ * @apiSuccess  {Integer}   id                    Id of the manifest
+ * @apiSuccess  {String}    name                  Name of the manifest
+ * @apiSuccess  {String}    creation_date         Date of creation
+ * @apiSuccess  {String}    last_update           Date of last update
+ * @apiSuccess  {Object[]}  history               List of content (last updated first)
```
Comment (Arignir) [inline server/src/action/manifest/get.js]:
```suggestion
 * @apiSuccess  {Object[]}  history               List of content (ordered by last_update)
```

File: server/src/action/manifest/get.js [hunk 10]
```diff
@@ -1,6 +1,66 @@
...
+ * @apiSuccess  {Integer}   history.id            Id of the content
+ * @apiSuccess  {Integer}   history.manifest_id   Id of the manifest
+ * @apiSuccess  {String}    history.content       The content of the manifest
+ * @apiSuccess  {String}    history.edition_date  Date of content edition
```
Comment (Arignir) [inline server/src/action/manifest/get.js]:
```suggestion
 * @apiSuccess  {String}    history.edition_date  Date when the manifest was edited
```

File: server/src/action/manifest/get.js [hunk 11]
```diff
@@ -1,6 +1,66 @@
...
+ * @apiParam  (Params) {Integer}  id  Id of the manifest
+ *
+ * @apiSuccess  {Integer}   id                    Id of the manifest
+ * @apiSuccess  {String}    name                  Name of the manifest
+ * @apiSuccess  {String}    creation_date         Date of creation
+ * @apiSuccess  {String}    last_update           Date of last update
+ * @apiSuccess  {Object[]}  history               List of content (last updated first)
+ * @apiSuccess  {Integer}   history.id            Id of the content
+ * @apiSuccess  {Integer}   history.manifest_id   Id of the manifest
+ * @apiSuccess  {String}    history.content       The content of the manifest
```
Comment (Arignir) [inline server/src/action/manifest/get.js]:
```suggestion
 * @apiSuccess  {String}    history.content       Content of an edition of the manifest 
```

File: server/src/action/manifest/get.js [hunk 12]
```diff
@@ -1,6 +1,66 @@
...
+ * @apiSuccess  {String}    history.edition_date  Date of content edition
+ *
+ * @apiSuccessExample {json} Response
+ * {
+ *    "id": 56,
+ *    "name": "to_compile",
```
Comment (Arignir) [inline server/src/action/manifest/get.js]: Same here

File: server/src/action/manifest/list.js [hunk 13]
```diff
@@ -3,6 +3,66 @@ const { query } = require('express-validator/check')
...
 
+/**
+ * @api {get} /api/manifest List
+ * @apiVersion 1.0.0
+ * @apiGroup Manifest
+ * @apiName ManifestList
+ *
+ * @apiDescription List all manifests
+ *
+ * @apiParam  (Query String) {String}                             [name]              Search results starting by provided name
```
Comment (Arignir) [inline server/src/action/manifest/list.js]:
```suggestion
 * @apiParam  (Query String) {String}                             [name]              Search for manifests with a name starting with `name`
```

File: server/src/action/manifest/list.js [hunk 14]
```diff
@@ -3,6 +3,66 @@ const { query } = require('express-validator/check')
...
+ * @apiParam  (Query String) {String='creation','update','name'}  [sort='creation']   Sort by provided column
```
Comment (Arignir) [inline server/src/action/manifest/list.js]:
```suggestion
 * @apiParam  (Query String) {String='creation','update','name'}  [sort='creation']   Sort the results by the provided criteria
```

File: server/src/action/manifest/list.js [hunk 15]
```diff
@@ -3,6 +3,66 @@ const { query } = require('express-validator/check')
...
+ * @apiParam  (Query String) {String='asc','desc'}                [dir='desc']        Direction of the sorting
```
Comment (Arignir) [inline server/src/action/manifest/list.js]:
```suggestion
 * @apiParam  (Query String) {String='asc','desc'}                [dir='desc']        Sort direction
```

File: server/src/action/manifest/list.js [hunk 16]
```diff
@@ -3,6 +3,66 @@ const { query } = require('express-validator/check')
...
+ * @apiSuccess  {Integer}   data.history.id             Id of the content
+ * @apiSuccess  {Integer}   data.history.manifest_id    Id of the manifest
+ * @apiSuccess  {String}    data.history.content        The content of the manifest
+ * @apiSuccess  {String}    data.history.edition_date   Date of content edition
+ * @apiSuccess  {Object}    meta                        Metadata of the paginated result
+ * @apiSuccess  {Object}    meta.pagination             Pagination information
+ * @apiSuccess  {Integer}   meta.pagination.total       Total number of item found
+ * @apiSuccess  {Integer}   meta.pagination.perPage     Number of item per page
+ * @apiSuccess  {Integer}   meta.pagination.currentPage Current page
+ * @apiSuccess  {Integer}   meta.pagination.pageCount   Number of pages containing results
```
Comment (Arignir) [inline server/src/action/manifest/list.js]:
```suggestion
 * @apiSuccess  {Integer}   meta.pagination.pageCount   Total number of pages
```

File: server/src/action/manifest/update.js [hunk 17]
```diff
@@ -2,6 +2,58 @@ const AbstractAction = require('../abstract')
...
+/**
+ * @api {put} /api/manifest/:id Update
+ * @apiVersion 1.0.0
+ * @apiGroup Manifest
+ * @apiName ManifestUpdate
+ *
+ * @apiDescription Update a manifest
+ *
+ * @apiParam  (Params) {Integer} id      Id of the manifest
+ * @apiParam  (Body)   {String{1..}}  content Content of the manifest
```
Comment (Arignir) [inline server/src/action/manifest/update.js]:
```suggestion
 * @apiParam  (Body)   {String{1..}}  content New content of the manifest
```

File: server/src/action/manifest/create.js [hunk 18]
```diff
@@ -2,6 +2,67 @@ const AbstractAction = require('../abstract')
...
+ *
+ * @apiDescription Create a manifest to compile
+ *
+ * @apiParam  (Body) {String{1..255}}  name    Name of the manifest
+ * @apiParam  (Body) {String{1..}}  content Content of the manifest
+ *
+ * @apiParamExample {json} Request Body
+ * {
+ *    "name": "bonjour",
+ *    "content": "{{content of a python file manifest}}"
```
Comment (Arignir) [inline server/src/action/manifest/create.js]:
```suggestion
 *    "content": "{{content of a build manifest (python)}}"
```

**Review by Arignir — CHANGES_REQUESTED** [review]
I'm counting on you to apply the suggested modifications on all routes

**Review by person-m — DISMISSED** [review]

**Review by Arignir — DISMISSED** [review]

**Review by Arignir — APPROVED** [review]

**Pull request merged**